src/Microsoft.ML.FastTree/Dataset/IntArray.cs

```diff
@@ -199,7 +199,7 @@ public static IntArray New(byte[] buffer, ref int position)
         /// <returns>An indexer into the array</returns>
         public abstract IIntArrayForwardIndexer GetIndexer();
 
-        public virtual void Sumup(SumupInputData input, FeatureHistogram histogram)
+        internal virtual void Sumup(SumupInputData input, FeatureHistogram histogram)
```

Comment:
>internal [](start = 8, length = 8)

OK, I see you made it internal here in the base class. Is it necessary to make it internal? I think let it be public because class is going to be internal anyway.

Reply:
I believe I was getting an error for leaving it as public when the class was internal, may be because its virtual?

---
In reply to: [257370838](https://github.com/dotnet/machinelearning/pull/2511#discussion_r257370838) [](ancestors = 257370838)

Reply:
Hi @codemzs could you look into this? We *definitely* have elsewhere internal classes with public virtual and abstract methods, the idea that this is inherently problematic makes no sense to me. Something else was going wrong. So I hope you don't mind, just going to reactivate, since as @zeahmed says we generally prefer to not do this.

---
In reply to: [257425227](https://github.com/dotnet/machinelearning/pull/2511#discussion_r257425227) [](ancestors = 257425227,257370838)

Reply:
Done.

---
In reply to: [258231590](https://github.com/dotnet/machinelearning/pull/2511#discussion_r258231590) [](ancestors = 258231590,257425227,257370838)